From: Anthony Liguori <aliguori@us.ibm.com>
Subject: SLES9 installation: Try starting linuxrc to allow starting an install

Xen currently implements the ramdisk option which seems to simply pull
in a gzip'd file to a ramdisk and set that as the root.  Looking at the
initrd code in the kernel and reading through the man page, initrd is a
bit more complicated than that.

Among other things, the initrd process will check for a /linuxrc file
and execute it.  After execution, it will remount a new root and then
fall through to the normal init process.

Acked-by: garloff@suse.de

diff -uNrp linux-2.6.10.xen/init/main.c linux-2.6.10.xen2/init/main.c
--- linux-2.6.10.xen/init/main.c	2005-01-21 19:45:10.986945023 +0100
+++ linux-2.6.10.xen2/init/main.c	2005-01-21 19:54:29.458106293 +0100
@@ -722,6 +722,7 @@ static int init(void * unused)
 	if (execute_command)
 		run_init_process(execute_command);
 
+	run_init_process("/linuxrc");
 	run_init_process("/sbin/init");
 	run_init_process("/etc/init");
 	run_init_process("/bin/init");
